File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    answer = -1
    stack = []
    i = 0 
    while i != len(s):
        if not stack:
            stack.append(s[i])
        elif stack[-1] == s[i]:
            stack.pop()
        else: 
            stack.append(s[i])
        i += 1
    if stack:
        return 0
    return 1

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int solution(string s)
{
    stack<int> st;
    
    for (int i = 0; i < s.size(); i++) {
        if (st.empty()){
            st.push(s[i]);
        }
        else if (st.top() == s[i]){
            st.pop();
        }
        else {
            st.push(s[i]);
        } 
    }
    return st.empty();
}